testutil/nullify/nullify.go

```diff
@@ -0,0 +1,57 @@
+// Package nullify provides methods to init nil value structs for test assertion.
+package nullify
+
+import (
+	"reflect"
+	"unsafe"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var (
+	coinType  = reflect.TypeOf(sdk.Coin{})
+	coinsType = reflect.TypeOf(sdk.Coins{})
+)
+
+// Fill analyzes all struct fields and slices with
+// reflection and initializes the nil and empty slices,
+// structs, and pointers.
+func Fill(x interface{}) interface{} {
```

Comment:
[Non Blocking]
I don't like having unsafe code on the state machine side. Isn't there a way not to use this?

Reply:
Agreed. The only usage of `Fill` appears to be in a test and I think the test can remove its usage ref: https://github.com/celestiaorg/celestia-app/pull/893#discussion_r1004230693 so if possible, I think we should remove this file entirely.

Reply:
Done